### ISSUE TITLE
UIImage extensions not exposed in ObjC

### ISSUE THREAD

**Issue by tielur** [zenangst/Hue#14]
I've installed Hue via CocoaPods and I'm trying to use the [UIImage color](https://github.com/hyperoslo/Hue/blob/master/Source/iOS/UIImage%2BHue.swift#L24) extension within an Objective-C `ViewController`. But looking at the header file exposed it doesn't seem to expose any of the `UIImage` extensions. At first I thought this may be related to the [UIImage extension not being public](https://github.com/hyperoslo/Hue/blob/master/Source/iOS/UIImage%2BHue.swift#L13) compared to the [UIColor extension](https://github.com/hyperoslo/Hue/blob/master/Source/iOS/UIColor%2BHue.swift#L5). So I forked that and changed the extension to be public but that didn't seem to help.

I'm not that familiar with using Swift within an Objective-C project, but according to a little research and the [Apple docs](https://developer.apple.com/library/ios/documentation/Swift/Conceptual/BuildingCocoaApps/MixandMatch.html#//apple_ref/doc/uid/TP40014216-CH10-ID122) this should be possible.

Thoughts?

`Hue-Swift.h`:  The `@interface UIImage` is near the bottom.

``` ObjC
// Generated by Apple Swift version 2.1.1 (swiftlang-700.1.101.15 clang-700.1.81)
#pragma clang diagnostic push

#if defined(__has_include) && __has_include(<swift/objc-prologue.h>)
# include <swift/objc-prologue.h>
#endif

#pragma clang diagnostic ignored "-Wauto-import"
#include <objc/NSObject.h>
#include <stdint.h>
#include <stddef.h>
#include <stdbool.h>

#if defined(__has_include) && __has_include(<uchar.h>)
# include <uchar.h>
#elif !defined(__cplusplus) || __cplusplus < 201103L
typedef uint_least16_t char16_t;
typedef uint_least32_t char32_t;
#endif

typedef struct _NSZone NSZone;

#if !defined(SWIFT_PASTE)
# define SWIFT_PASTE_HELPER(x, y) x##y
# define SWIFT_PASTE(x, y) SWIFT_PASTE_HELPER(x, y)
#endif
#if !defined(SWIFT_METATYPE)
# define SWIFT_METATYPE(X) Class
#endif

#if defined(__has_attribute) && __has_attribute(objc_runtime_name)
# define SWIFT_RUNTIME_NAME(X) __attribute__((objc_runtime_name(X)))
#else
# define SWIFT_RUNTIME_NAME(X)
#endif
#if defined(__has_attribute) && __has_attribute(swift_name)
# define SWIFT_COMPILE_NAME(X) __attribute__((swift_name(X)))
#else
# define SWIFT_COMPILE_NAME(X)
#endif
#if !defined(SWIFT_CLASS_EXTRA)
# define SWIFT_CLASS_EXTRA
#endif
#if !defined(SWIFT_PROTOCOL_EXTRA)
# define SWIFT_PROTOCOL_EXTRA
#endif
#if !defined(SWIFT_ENUM_EXTRA)
# define SWIFT_ENUM_EXTRA
#endif
#if !defined(SWIFT_CLASS)
# if defined(__has_attribute) && __has_attribute(objc_subclassing_restricted) 
#  define SWIFT_CLASS(SWIFT_NAME) SWIFT_RUNTIME_NAME(SWIFT_NAME) __attribute__((objc_subclassing_restricted)) SWIFT_CLASS_EXTRA
#  define SWIFT_CLASS_NAMED(SWIFT_NAME) __attribute__((objc_subclassing_restricted)) SWIFT_COMPILE_NAME(SWIFT_NAME) SWIFT_CLASS_EXTRA
# else
#  define SWIFT_CLASS(SWIFT_NAME) SWIFT_RUNTIME_NAME(SWIFT_NAME) SWIFT_CLASS_EXTRA
#  define SWIFT_CLASS_NAMED(SWIFT_NAME) SWIFT_COMPILE_NAME(SWIFT_NAME) SWIFT_CLASS_EXTRA
# endif
#endif

#if !defined(SWIFT_PROTOCOL)
# define SWIFT_PROTOCOL(SWIFT_NAME) SWIFT_RUNTIME_NAME(SWIFT_NAME) SWIFT_PROTOCOL_EXTRA
# define SWIFT_PROTOCOL_NAMED(SWIFT_NAME) SWIFT_COMPILE_NAME(SWIFT_NAME) SWIFT_PROTOCOL_EXTRA
#endif

#if !defined(SWIFT_EXTENSION)
# define SWIFT_EXTENSION(M) SWIFT_PASTE(M##_Swift_, __LINE__)
#endif

#if !defined(OBJC_DESIGNATED_INITIALIZER)
# if defined(__has_attribute) && __has_attribute(objc_designated_initializer)
#  define OBJC_DESIGNATED_INITIALIZER __attribute__((objc_designated_initializer))
# else
#  define OBJC_DESIGNATED_INITIALIZER
# endif
#endif
#if !defined(SWIFT_ENUM)
# define SWIFT_ENUM(_type, _name) enum _name : _type _name; enum SWIFT_ENUM_EXTRA _name : _type
#endif
typedef float swift_float2  __attribute__((__ext_vector_type__(2)));
typedef float swift_float3  __attribute__((__ext_vector_type__(3)));
typedef float swift_float4  __attribute__((__ext_vector_type__(4)));
typedef double swift_double2  __attribute__((__ext_vector_type__(2)));
typedef double swift_double3  __attribute__((__ext_vector_type__(3)));
typedef double swift_double4  __attribute__((__ext_vector_type__(4)));
typedef int swift_int2  __attribute__((__ext_vector_type__(2)));
typedef int swift_int3  __attribute__((__ext_vector_type__(3)));
typedef int swift_int4  __attribute__((__ext_vector_type__(4)));
#if defined(__has_feature) && __has_feature(modules)
@import UIKit;
@import CoreGraphics;
#endif

#pragma clang diagnostic ignored "-Wproperty-attribute-mismatch"
#pragma clang diagnostic ignored "-Wduplicate-method-arg"

@interface UIColor (SWIFT_EXTENSION(Hue))
+ (UIColor * __nonnull)hex:(NSString * __nonnull)string;
- (UIColor * __nonnull)colorWithMinimumSaturation:(CGFloat)minSaturation;
- (UIColor * __nonnull)alpha:(CGFloat)value;
@end


@interface UIColor (SWIFT_EXTENSION(Hue))
- (NSString * __nonnull)hexWithPrefix:(BOOL)withPrefix;
@property (nonatomic, readonly) BOOL isDark;
@property (nonatomic, readonly) BOOL isBlackOrWhite;
- (BOOL)isDistinctFrom:(UIColor * __nonnull)color;
- (BOOL)isContrastingWith:(UIColor * __nonnull)color;
@end


@interface UIImage (SWIFT_EXTENSION(Hue))
@end

#pragma clang diagnostic pop

```

**Comment by tielur** [comment]
So after digging a little bit more I've found that tuples are not supported in ObjC.

**Comment by zenangst** [comment]
Hey @tielur, we don't aim to support Objective-C with this library. We just feel that if we start doing that it is gonna be hard draw a line where Objective-C support ends and Swift enhancement begins. Hope that made sense for anyone else reading this because it made sense in my head :sunglasses: 

tl;dr
Adding Objective-C support is holding back development with Swift.
